### PR TITLE
Fixed a bug to continue visting NextPageLink when listing key vaults from ARM API

### PR DESCRIPTION
**Pull request by BethanyZhou**
- Fixed a bug to continue visting NextPageLink when listing key vaults from ARM API 
- Avoid listing key vaults twice

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs [hunk 1]
```diff
@@ -161,22 +161,36 @@ protected List<PSKeyVaultIdentityItem> ListVaults(string resourceGroupName, Hash
             return vaults;
         }
 
-        public virtual IEnumerable<PSKeyVaultIdentityItem> ListResources(Rest.Azure.OData.ODataQuery<GenericResourceFilter> filter = null, ulong first = ulong.MaxValue, ulong skip = ulong.MinValue)
+        public virtual IEnumerable<PSKeyVaultIdentityItem> ListResources(Rest.Azure.OData.ODataQuery<GenericResourceFilter> filter = null)
         {
             IResourceManagementClient armClient = ResourceClient;
-
-            return new GenericPageEnumerable<GenericResource>(() => armClient.Resources.List(filter), armClient.Resources.ListNext, first, skip).Select(r => new PSKeyVaultIdentityItem(r));
+            var response = armClient.Resources.List(filter);
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs:167]: Refactor to avoid duplicated code

Comment (isra-fel) [reply src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs:167]: Introduce a new method
`ListPagable(Func<type of response> getFirstPage, Func<string, type of response> getNextPage)`
Then you can simplify these two methods to call `ListPagable`

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/KeyVault/KeyVault/ChangeLog.md [hunk 2]
```diff
@@ -18,7 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
-
+* Fixed a bug to continue visting `NextPageLink` when listing key vaults from ARM API
```
[the remark this answers is not in the document]
Comment (BethanyZhou) [reply src/KeyVault/KeyVault/ChangeLog.md:21]: No bug id, this issue is reported by email

File: src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs (hunk 1 above)
Comment (BethanyZhou) [reply src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs]: Refactored without delegate function as the signature of Resources.List(filter) and ResourceGroup.ListResources(resourcegroupName, filter) is different.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**